src/html-entrypoint.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const { tokenize } = require('simple-html-tokenizer');
+const fs = require('fs');
+const path = require('path');
+const pattern = /<fastboot-script[^>]*><\/fastboot-script>\n*/g;
+
+function htmlEntrypoint(distPath, htmlPath) {
+  let html = fs.readFileSync(path.join(distPath, htmlPath), 'utf8');
+
+  // all the scripts we want to run go into appFiles. We don't use vendorFiles.
+  // The distinction doesn't matter here, as long as the scripts all stay in the
+  // right relative order.
+  let appFiles = [];
+
+  for (let token of tokenize(html)) {
+    if (token.type === 'StartTag' && ['script', 'fastboot-script'].includes(token.tagName)) {
+      for (let [name, value] of token.attributes) {
+        if (name === 'src') {
+          appFiles.push(path.join(distPath, value));
+        }
+      }
+    }
+  }
```

Comment:
FWIW, @stefanpenner wrote https://github.com/stefanpenner/find-scripts-srcs-in-document specifically for this use case.

Reply:
Yes, but  that library doesn't actually meet the requirements here (we need to control the set of tag names it matches), and we're talking about an 18 line function that is used by exactly zero other packages.

Reasonable people can disagree on the "lots of tiny libraries" approach. For me, this falls far, far below the bar of deserving to be its own library. It's hyper-specialized for this one use case. People doing anything more general will reach for a complete parser instead.

Reply:
The library was only created so that we could add [proper tests](https://github.com/stefanpenner/find-scripts-srcs-in-document/blob/master/test.js) for various combinations. Seems 100% fine to have this logic inlined, but we need to beef up the test coverage for the parsing bit.